internal/controller/services/monitoring/monitoring_controller_support.go

```diff
@@ -29,59 +33,76 @@ func getTemplateData(ctx context.Context, rr *odhtypes.ReconciliationRequest) (m
-			memoryLimit = metrics.Resources.MemoryLimit.String()
-			cpuRequest = metrics.Resources.CPURequest.String()
-			memoryRequest = metrics.Resources.MemoryRequest.String()
-		} else { // here need to match default value set in API
-			cpuLimit = "500m"
-			memoryLimit = "512Mi"
-			cpuRequest = "100m"
-			memoryRequest = "256Mi"
+			templateData["CPULimit"] = defaultIfEmpty(metrics.Resources.CPULimit.String(), "500m")
+			templateData["MemoryLimit"] = defaultIfEmpty(metrics.Resources.MemoryLimit.String(), "512Mi")
+			templateData["CPURequest"] = defaultIfEmpty(metrics.Resources.CPURequest.String(), "100m")
+			templateData["MemoryRequest"] = defaultIfEmpty(metrics.Resources.MemoryRequest.String(), "256Mi")
+		} else {
+			// Use defaults when Resources is nil
+			templateData["CPULimit"] = "500m"
+			templateData["MemoryLimit"] = "512Mi"
+			templateData["CPURequest"] = "100m"
+			templateData["MemoryRequest"] = "256Mi"
 		}
 
-		var storageSize, storageRetention string
+		// Handle Storage fields - provide defaults if Storage is nil
 		if metrics.Storage != nil {
-			storageSize = metrics.Storage.Size.String()
-			storageRetention = metrics.Storage.Retention
-		} else { // here need to match default value set in API
-			storageSize = "5Gi"
-			storageRetention = "1d"
+			templateData["StorageSize"] = defaultIfEmpty(metrics.Storage.Size.String(), "5Gi")
+			templateData["StorageRetention"] = defaultIfEmpty(metrics.Storage.Retention, "1d")
+		} else {
+			// Use defaults when Storage is nil
+			templateData["StorageSize"] = "5Gi"
+			templateData["StorageRetention"] = "1d"
 		}
 
+		templateData["MonitoringStackName"] = MSName
+
 		// only when either storage or resources is set, we take replicas into account
 		// - if user did not set it / zero-value "0", we use default value of 2
 		// - if user set it to Y, we pass Y to template
 		var replicas int32 = 2 // default value to match monitoringstack CRD's default
 		if (metrics.Storage != nil || metrics.Resources != nil) && metrics.Replicas != 0 {
 			replicas = metrics.Replicas
 		}
-
-		templateData["CPULimit"] = cpuLimit
-		templateData["MemoryLimit"] = memoryLimit
-		templateData["CPURequest"] = cpuRequest
-		templateData["MemoryRequest"] = memoryRequest
-		templateData["StorageSize"] = storageSize
-		templateData["StorageRetention"] = storageRetention
 		templateData["Replicas"] = strconv.Itoa(int(replicas))
 	}
 
-	// Add traces data if traces are configured
-	if monitoring.Spec.Traces != nil {
-		traces := monitoring.Spec.Traces.Storage
-		templateData["Backend"] = traces.Backend
-		templateData["Secret"] = traces.Secret
-		templateData["Size"] = traces.Size
+	// Add traces-related data if traces are configured
+	if traces := monitoring.Spec.Traces; traces != nil {
+		templateData["InstrumentationName"] = InstrumentationName
+		templateData["OtlpEndpoint"] = fmt.Sprintf("http://otel-collector.%s.svc.cluster.local:4317", monitoring.Spec.Namespace)
```

Comment:
After renaming this should be
```suggestion
		templateData["OtlpEndpoint"] = fmt.Sprintf("http://data-science-collector.%s.svc.cluster.local:4317", monitoring.Spec.Namespace)
```